src/SearchWrapper.cpp

```diff
@@ -0,0 +1,163 @@
+#include <Rcpp.h>
+#include <xapian.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+Rcpp::List convertToDataFrame(Rcpp::List list) {
+  Rcpp::List dataFrame = clone(list);
+  Rcpp::GenericVector aRow = dataFrame(0);
+  Rcpp::StringVector rowNames(aRow.length());
+  
+  for (int i = 0; i < aRow.length(); ++i) {
+    char name[5];
+    sprintf(&(name[0]), "%d", i+1);
```

Comment:
You can be a little clearer here than `&(name[0])`, because just `name` will do the job :)

Also, you shouldn't use `sprintf()`; use `snprintf()` instead, which is guaranteed not to overflow. (Even in situations where it can't happen in practice, it's better to use `snprintf()` so that you get into the habit, and so other people looking at your code don't worry it could have problems.